### ISSUE TITLE
Quarkus 3.8.1: Use GraalVM sdk 23.1.2 over 23.0.1

### ISSUE THREAD

**Issue by jerboaa** [quarkusio/quarkus#39257]
### Describe the bug

The bom for Quarkus 3.8 is supposed to be used with GraalVM for JDK 21 (internal version 23.1). However the GraalVM SDK version is still `23.0.1`. It should be `23.1.2` from [maven central](https://mvnrepository.com/artifact/org.graalvm.sdk/graal-sdk/23.1.2).

See:
https://github.com/quarkusio/quarkus/blob/3.8.1/bom/application/pom.xml#L92

The default image seems fine, pointing to the JDK 21 image. See:
https://github.com/quarkusio/quarkus/blob/3.8.1/core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java#L24

**Comment by gsmet** [comment]
@jerboaa yeah, I agree it needs fixing. It's harder for us to keep track of this though with the new version scheme we have for GraalVM images.
So if you're positive this is the version we should use, please create a PR. Thanks!

**Comment by maxandersen** [comment]
I'm surprised we use "jdk-21" and not the versions that has both the graalvm and jdk version present from https://quay.io/repository/quarkus/ubi-quarkus-mandrel-builder-image?tab=tags

@cescoffier @zakkak wasn't the combi numbers what we were after using?

**Comment by jerboaa** [comment]
> So if you're positive this is the version we should use, please create a PR. Thanks!

https://github.com/quarkusio/quarkus/pull/39260 here you go. That means the minimum JDK requirement is 17 (the old one, `23.0.1` used to be JDK 11+)

**Comment by maxandersen** [comment]
Just that the maven artifacts is using the old.

**Comment by geoand** [comment]
@jerboaa that's not the workflow we follow 😉 

The issue is closed and the milestone set to `3.9`, but that will automatically be updated when the PR that closed it is backported.

**Comment by jerboaa** [comment]
@geoand OK. Not familiar with the workflow, sorry ;-) As long as it ends up in the 3.8 branch I'm happy.

**Comment by geoand** [comment]
👌🏼
